map/tripManager.swift

```diff
@@ -8,38 +8,46 @@
 
 import Foundation
 
-class TripManager: gitProtocol
-{
+class TripManager: gitProtocol {
+    
+    /*
+     * TODO: implementaion MULTI THREADING
+     */
+    
+    static let shared = TripManager()
+    
     var trips = [Trip]()
     var currentTrip:Trip?
     var currentEvent:commit?
     
-    var tracker:routeTracker
-    let monitor:Monitor
+    private init() {
+        
+    }
     
-    func createCommit(type:eventType, with member:[Member], media:Media? = nil) -> commit? {
-        var event:commit?
-        switch type {
-        case .start:
-            event = Start(member)
-        case .route:
-            event = Route(member)
-        case .end:
-            event = End(member)
-        case .pause:
-            event = Pause(member)
-        case .resume:
-            event = Resume(member)
-        case .photo:
-            event = Photo(member, media: media as! photo)
-        default:
-            print("\(type) is not implimented")
-        }
-        if let event = event {
-            return event
-        } else {
-            return nil
-        }
+    func createCommit(type: eventType, with member: [Member], media: Media?) -> commit? {
+        let event: commit?
+        
+        event = {
```

Comment:
what's differ from before?

Reply:
It's intended to do upcasting from descendants(Start, Route, ...) to its super class Commit.
But not sure really. Might need to be changed.

Reply:
Oh, and I temporary changed define keyword "var" to "let" and assigned a value looks more clearly.